codec-base/src/test/java/io/netty/handler/codec/ByteToMessageDecoderTest.java

```diff
@@ -684,4 +691,109 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
         assertEquals(0, buffer.refCnt(), "Buffer should be released");
         assertFalse(channel.finish());
     }
+
+    @Test
+    void reentrantReadSafety() throws Exception {
+        LocalChannel channel = new LocalChannel();
```

Comment:
I used the local transport for the test as this will not work for `EmbeddedChannel` as explained in the commit message. Same for the other test below.